addons/website_event_track/static/src/js/website_event_track_proposal.js

```diff
@@ -0,0 +1,182 @@
+odoo.define('website_event_track.website_event_track_proposal', function (require) {
+    'use strict';
+
+    var core = require('web.core');
+    var publicWidget = require('web.public.widget');
+    var utils=require('web.utils');
+
+    var _t = core._t;
+
+    publicWidget.registry.websiteEventProposalForm = publicWidget.Widget.extend({
+
+    selector: '.o_website_event_track_proposal_form',
+    events: {
+        'input input[name="partner_name"]': '_onNameInputPropagateOnContact',
+        'click .o_wetrack_add_contact_information': '_onToggleContactInformationClick',
+        'click .o_wetrack_proposal_submit_button': '_send',
+    },
+
+    /**
+     * Check validity of input fields : empty required fields or bad formatted ones are 
+     * marked as is-invalid and o_has_error.
+     * 
+     * @private
+     */
+    _check_error_fields: function () {
+        var form_valid = true;
+
+        this.$('.form-group').each(function (k, field) { 
+            var $field = $(field);
+
+            // Validate inputs for this field, except if it is a select2 field.
+            var inputs = $field.find('.form-control').not('.o_wetrack_select2_tags');
+            var invalid_inputs = inputs.toArray().filter(function (input, k) {
+                return !input.checkValidity();
+            });
+            
+            // Add error display and classes to invalid fields.
+            $field.removeClass('o_has_error').find('.form-control').removeClass('is-invalid');
+            if (invalid_inputs.length) {
+                $field.addClass('o_has_error').find('.form-control').addClass('is-invalid');
+                form_valid = false;
+            }
+        });
+        return form_valid;
+    },
+
+    /**
+     * Handler for all desired behaviours on contact information when enabling/disabling contact information:
+     * - Toggle display of o_wetrack_contact_information section
+     * - Contact requirements (see _handleContactRequirements)
+     * - Error message / invalid fields displayed if no contact mean (email/phone) is entered. 
+     *   (removed if checkbox unticked)
+     * 
+     * @private
+     * @param {Event} ev
+     */
+    _onToggleContactInformationClick: function(ev){
+        var isChecked = $(ev.currentTarget).is(':checked');
+        var contactMeanError = this.$('#event_track_check_and_contact_information').hasClass('o_has_contact_mean_error');
+
+        if (isChecked) {
+            this.$('.o_wetrack_contact_information').removeClass('d-none');
+        } else {
+            this.$('.o_wetrack_contact_information').addClass('d-none');
+            // Empty the contact_email field if the entry is not valid. 
+            // Otherwise, the form is not sent and the error unnoticed.
+            if (!utils.is_email($('.o_wetrack_contact_email_input').val())){
```

Comment:
Actually it seems we could improve a bit this file by separating
  * UI
  * validation of form itself

For example here: clicking on "contact information" should just display / hide additional fields and change state of form in JS (useAdvancedContact for example). When validating the form at send event if useAdvancedContact is not checked you should simply filter out advanced information and skip it completely. That way you clearly separate "UI toggle" from "form content validation".

Reply:
Done!